pkg/gitnessclient/gitness_client.go

```diff
@@ -0,0 +1,289 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package gitnessclient
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+)
+
+const personalNamespaceId = "<PERSONAL>"
+
+type GitnessClient struct {
+	Token   string
+	BaseURL *url.URL
+}
+
+func NewGitnessClient(token string, baseUrl *url.URL) *GitnessClient {
+	return &GitnessClient{
+		Token:   token,
+		BaseURL: baseUrl,
+	}
+}
+
+// performRequest is the utility method to handle HTTP requests
```

Comment:
I'd remove this comment - doesn't add much information and other functions don't have them